src/yaml.cpp

```diff
@@ -105,6 +105,34 @@ estd::result<YAML::Node, estd::error> readYamlFile(std::string const & path) {
 	return YAML::Load(buffer.str());
 }
 
+estd::result<void, estd::error> mergeYamlNodes(YAML::Node map_a, YAML::Node map_b) {
+	// Check if the arguments are maps.
+	if (!map_a.IsMap() || !map_b.IsMap()) {
+		return estd::error{"invalid argument type"};
+	}
+
+	YAML::const_iterator iterator = map_b.begin();
+	while (iterator != map_b.end()) {
+		std::string key = iterator->first.as<std::string>();
+		YAML::Node value = iterator->second;
+		if (iterator->second.IsMap()) {
+			if (map_a[key]) {
```

Comment:
I'd recommend also testing if `map_a[key]` is a map. That allows you to generate an error with a nicer message than "invalid argument type" that the recursive call will generate.